src/PathReporter.ts

```diff
@@ -13,14 +14,14 @@ function getMessage(v: any, context: Context): string {
   return `Invalid value ${stringify(v)} supplied to ${getContextPath(context)}`
 }
 
-export function failure(es: Array<ValidationError>): Array<string> {
+export function failure(es: NonEmptyArray<ValidationError>): NonEmptyArray<string> {
   return es.map(e => getMessage(e.value, e.context))
 }
 
-export function success(): Array<string> {
-  return ['No errors!']
+export function success(): NonEmptyArray<string> {
+  return nonEmptyArray.of('No errors!')
 }
 
 export const PathReporter: Reporter<Array<string>> = {
-  report: validation => validation.fold(failure, success)
+  report: validation => validation.fold(failure, success).toArray()
```

Comment:
this is where I would love for this to be `Reporter<NonEmptyArray<string>>`, but it would be a breaking change, so let's talk about it.